Odata-docs/webapi-8/fundamentals/actions-functions.md

```diff
@@ -0,0 +1,423 @@
+    opt.RouteOptions.EnableNonParenthesisForEmptyParameterFunction = true;
+});
+```
+
+### Bound Action
+Actions are invoked via `POST` requests therefore we can pass a payload in the body.
+
+```json
+POST ~/odata/Books('1')/Rate
+{
+    "rating": 7
+}
+```
+
+### Unbound action
+```json
+POST ~/odata/incrementBookYear
+{
+    "increment": 7,
+    "id": "1"
+}
+```
+
+## Practical examples
+We will add detailed examples on how to configure bound and unbound functions and actions.
+We will start by setting up the ASP.NET Core OData v8.x project.
+
+### Data Model classes
+```csharp
+public class Book
+{
+    public string ID { get; set; }
+    public string Title { get; set; }
+    public bool ForKids { get; set; }
+}
+
+public class BookRating
+{
+    public string ID { get; set; }
+    public int Rating { get; set; }
+    public string BookID { get; set; }
+}
+```
+
+### Controller
+```csharp
+public class BooksController : ODataController
+{
+    // Get ~/Books
+    [EnableQuery]
+    public IActionResult Get()
+    {
+        return Ok(DataSource.Instance.Books);
+    }
+}
+```
+
+### Build Edm Model
+```csharp
+private static IEdmModel GetEdmModel()
+{
+    var builder = new ODataConventionModelBuilder();
+    builder.EntitySet<Book>("books");
+    var model = builder.GetEdmModel();
+    return model;
+}
+```
+
+## Bound Function
+Bound functions can be bound to an entity type or a collection of entity type.
+In our example model above, the bound function can be bound to a single book or a collection of books.
+We will demonstrated how to add a function that is bound to a collection.
+
+In the Edm Model, we modify the code as follows.
+
+```csharp
+private static IEdmModel GetEdmModel()
+{
+    var builder = new ODataConventionModelBuilder();
+    // -------------
+    // Existing code
+    // -------------
+
+    builder.EntityType<Book>().Collection
+        .Function("mostRecent")
+        .Returns<string>();
+    var model = builder.GetEdmModel();
+    return model;
+}
+```
+
+In the Edm Model above, we are adding a `mostRecent` function bound to the books collection. The function returns the `ID` of the most recent book. The `ID` is of `string` type.
+
+We will update the `BooksController` as follows.
+
+```csharp
+public class BooksController : ODataController
+{
+    // -------------
+    // Existing code
+    // -------------
+
+    [HttpGet("odata/Books/mostRecent()")]
+    public IActionResult MostRecent()
+    {
+        var maxBookId = DataSource.Instance.Books.Max(x => x.ID);
+        return Ok(maxBookId);
+    }
+}
+```
+
+If we invoke `GET odata/Books/mostRecent()`
+We get the response below:
+
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Edm.String",
+    "value": "8"
+}
+```
+
+## Unbound function
+Unbound functions don’t bind to any type and they are invoked as static operations. All unbound function overloads MUST have same return type.
```

Comment:
I think we should have a separate subsection for function overloads, so we can explain what overloads are, the limitations and provide examples. Same thing for actions.